Become changes

mtju opened the pull request:
A couple of small improvements regarding privilege escalation. Please take a look at commit messages for more information.

mtju committed:
Add stderr check for unpartitioned space check.

mtju committed:
Remove become from taks.

After taking a quick look at some of the most popular roles on Ansible Galaxy I could not find one with 'become' on a task level so I have removed it from expand filesystem task. Also I've updated example playbook.

charleskorn commented:
Thanks for another pull request. 

I don't have a Pi nearby to verify this myself... does `parted` need root access? The reason I ask is that I'd rather keep the `become` entries near the things that need root access - that way it's very clear why it's needed. (You'll notice `become` is also given explicitly in the restart handler as well for this reason.)

mtju committed:
Remove become from restart handler.

mtju committed:
Example playbook. Move become from playbook to individual role level.

Also add prefix for role name.

mtju commented:
Yes, `parted` requires root access. So everything requires root access except `Wait for restart to finish` handler. So this is what I have changed now:
- Removed `become` from restart handler since basically the whole playbook requires root. And I will also repeat myself but I am yet to find a popular role which uses `become` on task level.
- Updated example playbook with [role scoped become](https://stackoverflow.com/q/39183100/2374958).

charleskorn commented:
Looks good to me, thanks for the PR.
